crates/core/c8y_api/src/smartrest/smartrest_serializer.rs

```diff
@@ -32,185 +93,124 @@ impl From<CumulocitySupportedOperations> for &'static str {
+/// for statically supported operations.
+pub trait C8yOperation {
+    fn name(&self) -> &str;
 }
 
-impl Default for SmartRestGetPendingOperations {
-    fn default() -> Self {
-        Self { id: "500" }
+impl C8yOperation for CumulocitySupportedOperations {
+    fn name(&self) -> &str {
+        (*self).into()
     }
 }
 
-impl<'a> SmartRestSerializer<'a> for SmartRestGetPendingOperations {}
-
-#[derive(Debug, Deserialize, Serialize, Eq, PartialEq)]
-pub struct SmartRestSetOperationToExecuting {
-    pub message_id: &'static str,
-    pub operation: &'static str,
-}
-
-impl SmartRestSetOperationToExecuting {
-    pub fn new(operation: CumulocitySupportedOperations) -> Self {
-        Self {
-            message_id: "501",
-            operation: operation.into(),
-        }
+impl<'a> C8yOperation for &'a str {
+    fn name(&self) -> &str {
+        self
     }
 }
 
-impl<'a> SmartRestSerializer<'a> for SmartRestSetOperationToExecuting {}
-
-#[derive(Debug, Deserialize, Serialize, Eq, PartialEq)]
-pub struct SmartRestSetOperationToSuccessful {
-    pub message_id: &'static str,
-    pub operation: &'static str,
-    pub operation_parameter: Option<String>,
+#[derive(Debug, Serialize, Eq, PartialEq)]
+#[serde(untagged)]
+pub enum TextOrCsv {
+    Text(String),
+    Csv(EmbeddedCsv),
 }
 
-impl SmartRestSetOperationToSuccessful {
-    pub fn new(operation: CumulocitySupportedOperations) -> Self {
-        Self {
-            message_id: "503",
-            operation: operation.into(),
-            operation_parameter: None,
-        }
+impl From<EmbeddedCsv> for TextOrCsv {
+    fn from(value: EmbeddedCsv) -> Self {
+        Self::Csv(value)
     }
+}
 
-    pub fn with_response_parameter(self, response_parameter: &str) -> Self {
-        Self {
-            operation_parameter: Some(response_parameter.into()),
-            ..self
-        }
+impl From<String> for TextOrCsv {
+    fn from(value: String) -> Self {
+        Self::Text(value)
     }
 }
 
-impl<'a> SmartRestSerializer<'a> for SmartRestSetOperationToSuccessful {}
+#[derive(Debug, Eq, PartialEq)]
+pub struct EmbeddedCsv(String);
 
-#[derive(Debug, Deserialize, Serialize, Eq, PartialEq)]
-pub struct SmartRestSetOperationToFailed {
-    pub message_id: &'static str,
-    pub operation: &'static str,
-    #[serde(serialize_with = "reason_to_string_with_quotes")]
-    pub reason: String,
+impl EmbeddedCsv {
+    pub fn new(value: String) -> Self {
+        Self(value)
+    }
 }
 
-impl SmartRestSetOperationToFailed {
-    pub fn new(operation: CumulocitySupportedOperations, reason: String) -> Self {
-        Self {
-            message_id: "502",
-            operation: operation.into(),
-            reason,
+impl Serialize for EmbeddedCsv {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+        S::Error: serde::ser::Error,
+    {
+        let record = dbg!(parse_single_record(&self.0)?);
```

Comment:
Oops, didn't mean to leave that `dbg!` in. Just leaving this note here so I remember to get rid of it.